lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -449,24 +450,62 @@ def set_3d_properties(self, zs, zdir):
         self._edgecolor3d = self.get_edgecolor()
         self.stale = True
 
+    def set_facecolor(self, colors):
+        # override setter for facecolor so that 3dfacecolor
+        # will be updated as well. 
+        # In providing consitentcy in 2dcolor and 3dcolor setter.
+        # Note: will also update edgecolor if it follows facecolor
+
+        PathCollection.set_facecolor(self, colors)
+        self._facecolor3d = PathCollection.get_facecolor(self)
+        try:
+            # Will update edge color together if it follows facecolor
+            if (isinstance(self._edgecolors, six.string_types) and
+                self._edgecolors == str('face')):
+                    self.set_edgecolor(colors)
+        except AttributeError:
+            pass
+    set_facecolors = set_facecolor
+
+    def set_edgecolor(self, colors):
+        # override setter for edgecolor so that 3dedgecolor
+        # will be updated as well.
+        # In providing consitentcy in 2dcolor and 3dcolor setter.
+
+        PathCollection.set_edgecolor(self, colors)
+        self._edgecolor3d = PathCollection.get_edgecolor(self)
+    set_edgecolors = set_edgecolor
+
+    def _set_facecolors_on_project(self, colors):
```

Comment:
what is `project`? Should this rather be `_set_facecolors_on_projection` or `_set_facecolors_on_2d_projection`?